Messy.py

```diff
@@ -0,0 +1,252 @@
+
+        process = subprocess.Popen(build_command, shell=True)
+        process.wait()
+
+        # TODO: handle process status mor ecarefully: what happens if the file exists?
+        if os.path.isfile(apptainer_file):
+            print(f"Generated {apptainer_file}")
+            # print(process.stdout)
+            return True
+        else:
+            print("Image generation has failed.")
+            print(process.stderr)
+            return False
+
+    def check_command_exists(self, command: str) -> bool:
+        """Check whether `command` is on PATH and marked as executable."""
+        return which(command) is not None
+
+    def check_required_tools(self, command_list: list[str]) -> bool:
+        """Check whether all required commands are available."""
+
+        # TODO: also check the required versions
+
+        if all(self.check_command_exists(command) for command in command_list):
+            print("Have all the required external tools.")
+            return True
+        else:
+            return False
+
+    def generate_job_file(
+        self,
+        job_name: str,
+        sif_file_path: str,
+        model_path: str,
+        experiment_name: str,
+        table_name: str,
+        job_time: str,
+    ) -> None:
+        """Generate batch script file for job submission."""
+
+        environment = jinja2.Environment(
+            loader=jinja2.FileSystemLoader("templates/")
+        )
+        template = environment.get_template("netlogo_job.jinja")
+
+        filename = f"{job_name}.sh"
+        content = template.render(
+            job_name=job_name,
+            sif_file_path=sif_file_path,
+            job_time=job_time,
+            model_path=model_path,
+            experiment_name=experiment_name,
+            table_name=table_name,
+        )
+
+        with open(filename, mode="w", encoding="utf-8") as message:
+            message.write(content)
+            print(f"Batch job file was saved as {filename}")
+
+    def run_command_on_remote(
+        self, command: str, timeout: int = 5, flush: bool = False
+    ) -> None:
+        """Run a command on a remote system"""
+        transport = self.ssh_client.get_transport()
+        channel = transport.open_session()
+        channel.get_pty()
+        channel.settimeout(timeout)
+        channel.set_combine_stderr(True)
+        stdout = channel.makefile()
+        channel.exec_command(command)
+
+        stdout_lines = stdout.readlines()
+        for line in stdout_lines:
+            if flush:
+                print(line, end="", flush=True)
+            else:
+                print(line, end="")
+
+        # https://stackoverflow.com/a/11474509
+        if flush:
+            sys.stdout.write(
+                "\033[F" * len(stdout_lines)
+            )  # Cursor up for X number of lines
+
+    def watch_slurm_queue(self, sleep_time: float = 5.0) -> None:
+        """_summary_"""
+        # command = 'squeue --all'
+        command = "ls /home/xenon"
```

Comment:
squeue -j [job_id] seems a better command to monitor specific job.

Reply:
I created an issue to add another method.